middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -512,11 +508,15 @@ let close_let acc env id user_visible defining_expr
       (* CR pchambart: Not tail ! *)
       let acc, body = body acc body_env in
       match defining_expr with
-      | None -> acc, body
+      | None ->
+        (* If the defined variable has been seen at some point
+           it should have been replaced by now *)
+        let acc = Acc.remove_var_from_free_names var acc in
```

Comment:
I don't believe that's correct. Calls to `cont` with `None` can occur in `close_c_call`, in which case `var` is available from the `let_bound_var` parameter. This code generates its own `let` or `let_cont` bindings, and if you remove `var` here you risk passing incorrect free names information to these bindings.